sql/pgwire/encoding.go

```diff
@@ -78,7 +59,11 @@ func (b *readBuffer) readUntypedMsg(rd io.Reader) error {
 			size, maxMessageSize)
 	}
 
-	b.reset(size)
+	if len(b.msgBuf) < size {
```

Comment:
Not compelling at all. I think I'd drop this change, or replace with a comment that trying to reuse the buffer is both dangerous and not compelling from a performance perspective.